test/app_profiler/profile_server_test.rb

```diff
@@ -0,0 +1,332 @@
+# frozen_string_literal: true
+
+require "test_helper"
+require "net/http"
+
+module AppProfiler
+  module Server
+    TEST_PORT = 11337
+
+    class ServerTest < TestCase
+      def setup
+        Server.reset
+      end
+
+      test ".reset sets module variables back to their default" do
+        refute_equal(TRANSPORT_TCP, Server.transport)
+        Server.transport = TRANSPORT_TCP
+        refute_equal(DEFAULTS[:transport], Server.transport)
+        Server.reset
+        refute_equal(TRANSPORT_TCP, Server.transport)
+        assert_equal(DEFAULTS[:transport], Server.transport)
+      end
+    end
+
+    class TCPServerTest < TestCase
+      def setup
+        Server.reset
+        Server.transport = TRANSPORT_TCP
```

Comment:
Do we need to revert this assignment in this test case or the one below in a teardown? I don't think test cases should finish and leave global state changed. You can do something like

```
def with_transport(value)
  old_transport = Server.transport
  Server.transport = value
  yield
ensure
  Server.transport = old_transport
end

def run(*)
  with_transport(SOME_CONSTANT) do
    super
  end
end
```
or just use a teardown block.

Reply:
good point, i have been bit by this more than once already, I think doing this in a teardown block would be ideal (along with your superclass suggestion above too, I think).